subsys/net/lib/shell/iface.c

```diff
@@ -305,6 +328,16 @@ static void iface_cb(struct net_if *iface, void *user_data)
 	if (net_if_l2(iface) == &NET_L2_GET_NAME(ETHERNET)) {
 		PR("Ethernet capabilities supported:\n");
 		print_supported_ethernet_capabilities(sh, iface);
+
+#ifdef CONFIG_ETH_PHY_DRIVER
+		const struct device *phy_dev = net_eth_get_phy(iface);
+
+		PR("Ethernet PHY device: %s (%p)\n", (phy_dev != NULL) ? phy_dev->name : "<?>",
```

Comment:
I think there are some ethernet controllers where the phy is integrated with the mac, we should probably identify this case, and print that information accordingly rather than just "?" in those cases so that it doesn't look like we are in some unexpected state

Reply:
As said above I think these devices should have a separate `DEVICE_DEFINE` part for their integrated phy and implement the phy api functions in there. If that is done they would not be showing the `<?>`. For those who don't have a phy connected or implemented I think this would be the correct way. As this is done the same way as it is done for the device in line 270 (https://github.com/VOGL-electronic/sw__embedded_zephyr/blob/0522d63b452eee7df16818d6db4a1b1347344bb7/subsys/net/lib/shell/iface.c#L270)

Reply:
We could also move this line inside the condition and print nothing, when there is no phy.

Reply:
maybe instead of the "?", you can just print "none" or something like that to be clear about what is the situation, it would still have the effect of indicating possibly a wrong situation while also being expected if there really is no separate phy, as opposed to a menacing and mysterious "?"